lib/std/js/jsasynchttpclient.nim

```diff
@@ -0,0 +1,93 @@
+  text(await fetch(cstring($url), fetchOptionsImpl(body.cstring, "POST".cstring)))
+
+proc putContent*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[cstring] {.async.} =
+  text(await fetch(cstring($url), fetchOptionsImpl(body.cstring, "PUT".cstring)))
+
+proc patchContent*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[cstring] {.async.} =
+  text(await fetch(cstring($url), fetchOptionsImpl(body.cstring, "PATCH".cstring)))
+
+proc get*(self: JsAsyncHttpClient; url: Uri | string): Future[Response] {.async.} =
+  fetch(cstring($url))
+
+proc delete*(self: JsAsyncHttpClient; url: Uri | string): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl("".cstring, "DELETE".cstring))
+
+proc post*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl(body.cstring, "POST".cstring))
+
+proc put*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl(body.cstring, "PUT".cstring))
+
+proc patch*(self: JsAsyncHttpClient; url: Uri | string; body = ""): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl(body.cstring, "PATCH".cstring))
+
+proc head*(self: JsAsyncHttpClient; url: Uri | string): Future[Response] {.async.} =
+  fetch(cstring($url), fetchOptionsImpl("".cstring, "HEAD".cstring))
+
+
+runnableExamples("-d:nimExperimentalJsfetch -r:off"):
+  import std/asyncjs
+  from std/jsfetch import Response
+  from std/uri import parseUri, Uri
+
+  let client: JsAsyncHttpClient = newJsAsyncHttpClient()
+  const data: string = """{"key": "value"}"""
+
+  block:
+    proc example() {.async.} =
+      let url: Uri = parseUri("http://nim-lang.org")
+      let content: cstring = await client.getContent(url)
+      let response: Response = await client.get(url)
+    discard example()
+
+  block:
+    proc example() {.async.} =
+      let url: Uri = parseUri("http://httpbin.org/delete")
+      let content: cstring = await client.deleteContent(url)
+      let response: Response = await client.delete(url)
+    discard example()
+
+  block:
+    proc example() {.async.} =
+      let url: Uri = parseUri("http://httpbin.org/post")
+      let content: cstring = await client.postContent(url, data)
+      let response: Response = await client.post(url, data)
+    discard example()
+
+  block:
+    proc example() {.async.} =
+      let url: Uri = parseUri("http://httpbin.org/put")
+      let content: cstring = await client.putContent(url, data)
+      let response: Response = await client.put(url, data)
+    discard example()
+
+  block:
+    proc example() {.async.} =
+      let url: Uri = parseUri("http://httpbin.org/patch")
+      let content: cstring = await client.patchContent(url, data)
+      let response: Response = await client.patch(url, data)
+    discard example()
```

Comment:
@juancarlospaco do you know how to make top-level await work? I just heard of it today, eg see https://v8.dev/features/top-level-await

this would enable (if we can make it work) writing tests with better semantics, eg:
`await example()` instead of `discard example()`

Reply:
I think the support is too small yet to worth the efforts, maybe next year.
https://caniuse.com/mdn-javascript_operators_await_top_level